hashicorp/go-tfe
# CRUD support for no-code registry modules

## Description

This PR adds CRUD support for registry no-code modules. This will allow users to interact with the new no-code API. The no-code modules can be created with different variable options, and can be update as deemed fit.

## Testing plan


1. Create a User in TFC 
2. Create an Organization (e.g `my-org`) and ensure they have access to the new no-code API
3. Generate an API token for your user in TFC
4. In the tests in `registry_no_code_module_integration_test.go`, replace the org creation with a Read of the organization you created above. e.g
 ```go
	// orgTest, orgTestCleanup := createOrganization(t, client)
	// defer orgTestCleanup()
	orgTest, err := client.Organizations.Read(ctx, "my-org")
	if err != nil {
		t.Fatal(err)
	}
```
6. Replacing the placeholders with your TFC url and token, run the tests with 
```
ENABLE_BETA=1 TFE_ADDRESS="https://app.staging.terraform.io" TFE_TOKEN="your-api-token" go test -timeout 5m -run ^TestRegistryNoCodeModules github.com/hashicorp/go-tfe
```

## External links

- [JIRA](https://hashicorp.atlassian.net/browse/TFECO-3644)
